src/SmartComponents/Inventory/applications/compliance/SystemRulesTable.js

```diff
@@ -1,181 +1,364 @@
+            if (row.hasOwnProperty('parent') && row.parent > itemsPerPage) {
+                row.parent = row.parent % (itemsPerPage * 2);
+            }
 
-    onItemSelect = (_event, key, selected) => {
-        let { rows, page, itemsPerPage } = this.state;
-        const firstIndex = (page - 1) * (itemsPerPage * 2);
-        rows[firstIndex + Number(key)].selected = selected;
-        this.setState({
-            rows
+            return row;
+        });
+        const ruleIds = newRows.map((row) => {
+            if (row.hasOwnProperty('isOpen')) { return this.state.refIds[row.cells[0]]; }
+        }).filter(rule => rule).map(rule => 'compliance:' + rule);
+
+        return window.insights.chrome.auth.getUser()
+        .then(() => {
+            return fetch('/r/insights/platform/remediations/v1/resolutions', {
+                method: 'post',
+                headers: { 'Content-Type': 'application/json' },
+                body: JSON.stringify({ issues: ruleIds })
+            }).then((response) => {
+                if (!response.ok) {
+                    // If remediations doesn't respond, inject no fix available
+                    return {};
+                }
+
+                return response.json();
+            });
+        }).then(response => {
+            newRows = newRows.map((row) => {
+                if (row.hasOwnProperty('isOpen') && row.cells.length === 4) {
+                    if (response['compliance:' + this.state.refIds[row.cells[0]]]) {
+                        row.cells.push(<Ansible/>);
+                    } else {
+                        row.cells.push(<Ansible unsupported />);
+                    }
+                }
+
+                return row;
+            });
+
+            return newRows;
         });
     }
 
-    onExpandClick = (_event, _row, rowKey) => {
-        let { rows, page, itemsPerPage, openNodes } = this.state;
-        const activeRow = rows[(page - 1) * (itemsPerPage * 2) + Number(rowKey)];
-        activeRow.active = !activeRow.active;
-        if (!activeRow.active) {
-            openNodes.splice(openNodes.indexOf(Number(rowKey) + 1), 1);
+    onSelect(_event, selected, key) {
+        let { rows, page, itemsPerPage } = this.state;
+        if (key === -1) {
+            // This represents "Select all rules"
+            const lowerBound = (page - 1) * itemsPerPage * 2;
+            Array.from(new Array(itemsPerPage * 2), (x, i) => { rows[i + lowerBound].selected = selected; });
         } else {
-            openNodes = [
-                ...openNodes,
-                Number(rowKey) + 1
-            ];
+            // One rule was selected
+            rows[((page - 1) * itemsPerPage * 2) + Number(key)].selected = selected;
         }
 
-        this.setState({
-            openNodes,
-            rows
+        this.setState({ rows });
+    }
+
+    rulesToRows(profileRules) {
+        const rows = [];
+        const refIds = {};
+        profileRules.forEach((profileRule) => (
```

Comment:
You don't need to calculate colSpan and please do not use forEach, but rather return + flatMap
At top import `flatMap`
`import flatMap from 'lodash/flatMap'`
Do now use direct styles, these colors can't be changed by switching to dark/light theme, rather use classes.
Above this function
```JSX
calculateParent = ({ profile }, { title, severity, compliant }) => ({
  isOpen: false,
  cells: [
    title,
    profile,
    severity,
    (compliant ? <CheckCircleIcon className='ins-u-passed' /> :
      <ExclamationCircleIcon className='ins-u-failed' />)
  ]
})
```

Above this function
```JSX
calculateChild = ({ description, rationale }, key) => ({
  parent: key * 2,
  cells: [{
    title: (
      <React.Fragment key={ key }>
        <div id={`rule-description-${key}`}>
          <b>Description</b>
          <br/>
          <p>{ description }</p>
        </div>
        <br/>
        <div id={`rule-rationale-${key}`}>
          <b>Rationale</b>
          <br/>
          <p>{ rationale }</p>
        </div>
     </React.Fragment>
  }]
})
```

```JS
return flatMap(profileRules, (profileRule) => profileRule.map(rule, key) => {
  refIds[rule.title] = rule.ref_id;
  return [
    this.calculateParent(profileRule, rule),
    this.calculateChild(rule, key)
  ]
}));
```